tests/test_model.cpp

```diff
@@ -447,15 +456,18 @@ int test_model_sei_deterministic()
     config.set_date_end(2020, 12, 31);
     config.set_step_unit(StepUnit::Month);
     config.set_step_num_units(1);
+    config.use_mortality = false;
+    config.mortality_frequency = "year";
+    config.mortality_frequency_n = 1;
     config.create_schedules();
 
     config.deterministic = true;
 
     std::vector<std::vector<int>> movements;
 
-    unsigned num_mortality_years = config.num_mortality_years();
+    unsigned num_mortality_steps = 1;
```

Comment:
Is `config.num_mortality_years()` giving something else than 1? If yes, please note it here, we need to investigate or at least document it better. Or given you have `config.use_mortality = false;` above, shouldn't this be 0. That should work as mortality should not be touched with `use_mortality = false`. Both of these tests should work with `use_mortality = false` which is already set as the default in config. Maybe this is poorly defined in the API or impossible to do at this point because here we need 0 steps for something which optionally does not have a schedule. In any case, a lengthy but right and safe idiom would be:

```C++
unsigned num_mortality_steps = 0;
if (config.use_mortality)
    num_mortality_steps = config.num_mortality_steps();
```

...or in this case without the `if` part, only with a comment about not using mortality.

I think these two tests didn't have to care until now which would be nice to preserve, but perhaps with the above better expressed in the code.

Reply:
`config.num_mortality_steps()` is returning 0. Which is to be expected and correct behavior but to run the model we need mortality tracker to be of length 1 or greater. mortality_tracker is used in other modules even if `use_mortality=false` . So for the tests this should be 1 we could get rid of the num_mortality_steps = 1 and just use 1 to assign the mortality tracker.